### DISCUSSION TITLE
Jinx on NixOS

### DISCUSSION THREAD

**Discussion by patrl** [minad/jinx#104] [Q&A] [answered]
Thanks for the awesome package!

If someone tries install jinx in emacs while running NixOS, pkg-config won't find libenchant, even if the `enchant` and `pkg-config` packages are installed at the system level. This is due to a quirk of how header files work on NixOS (https://nixos.wiki/wiki/C). In order to successfully compile jinx, I had to write a `shell.nix` file with the following contents, and placed it in the jinx build dir:

``` nix
with import <nixpkgs> {};
stdenv.mkDerivation {
  name = "env";
  nativeBuildInputs = [ pkg-config ];
  buildInputs = [
    enchant
  ];
}
```

Entering into the nix shell then correctly sets the necessary environment variable for pkg-config, and jinx can be compiled using the script in `jinx.el`. In some sense the solution is very simple, but incorporating it into the package would involve an ad-hoc addition to the compilation script for NixOS, which I understand you may want to avoid. Some existing packages which use native compilation, such as pdf-tools, do exactly this - by using a more complicated build procedure which detects whether or not the user is running NixOS and enters into a shell as necessary; see, e.g., https://github.com/vedang/pdf-tools/blob/master/server/autobuild.

If you decide that this isn't a problem that the package should address, fair enough. Hopefully this will be useful to other NixOS users who encounter the same issue.

**Comment by minad** [comment]
Are you working on an official recipe for Nix? You may want to look at the corresponding Guix recipe in how they solved building and installation. I am not familiar enough to help.

**Reply by patrl** [reply, marked as answer]
Somehow it hadn't occurred to me to check, but it turns out that jinx is already packaged for NixOS (here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix). I usually just install emacs on NixOS and do package management in the traditional way, but indeed probably the cleanest solution is to just install packages like jinx using the system package manager. I was hoping to avoid installing only some packages this way, but it seems better than using a hack pdf-tools does. Cheers!

**Reply by madorian** [reply]
I still don't get how you make use of this file. How did you install jinx?

**Reply by fleimgruber** [reply]
@madorian https://wiki.nixos.org/wiki/Emacs#Installing_Packages
